helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -210,6 +213,18 @@ private void updateStateTransitionMessageThreadPool(Message message, HelixManage
       return;
     }
 
+    if (!_batchMessageThreadpoolChecked) {
```

Comment:
There is no HelixManger or ZkClient to read configurable information from ZK in constructor.